packages/core/src/storage/extract.ts

```diff
@@ -17,62 +22,96 @@ export function extractStorageLayout(
   contractDef: ContractDefinition,
   decodeSrc: SrcDecoder,
   deref: ASTDereferencer,
+  storageLayout?: StorageLayout | undefined,
 ): StorageLayout {
-  const layout: StorageLayout = { storage: [], types: {}, layoutVersion: currentLayoutVersion };
+  const layout: StorageLayout = { storage: [], types: {}, layoutVersion: currentLayoutVersion, flat: false };
+  if (storageLayout !== undefined) {
+    if (storageLayout.types) {
+      for (const key of Object.keys(storageLayout.types)) {
```

Comment:
I'm not sure I understand this. It feels like:

- `layout.types` is a key => { label, member } object.
- `layout.types` is almost a copy of `storageLayout.types`
- the difference it that, going from `storageLayout.types` to `layout.types` you apply a map
- the map really doesn't do anything ...

Reply:
The map does a `pick` to remove unwanted fields. In a previous comment that is now lost I suggested using the `pick` function directly. This should be:

```
solcMembers.map(m => typeof m === 'object' ? pick(m, ['label', 'type']) : m);